### PR TITLE
Ad738x: fix xilinx, merge projects, enable stm32 dma.

### PR DESCRIPTION
**Pull request by ahaslam2**
## Pull Request Description

This PR contains several fixes/improvements for the ad738x series:

-  fixes xilinx project not working because of missing pwm/clkgen,
- merge the stm32 and xilinx project, so that there is a single project for ad738x
- enable spi dma on stm32 for 1 Msps. 

## PR Type
- [x] Bug fix (change that fixes an issue)
- [x] New feature (change that adds new functionality)
- [ ] Breaking change (has dependencies in other repos or will cause CI to fail)

## PR Checklist
- [x] I have followed the [Coding style guidelines](http://analogdevicesinc.github.io/no-OS/drivers_guide.html#coding-style)
- [x] I have performed a self-review of the changes
- [x] I have commented my code, at least hard-to-understand parts
- [x] I have build all projects affected by the changes in this PR
- [x] I have tested in hardware affected projects, at the relevant boards
- [x] I have signed off all commits from this PR
- [ ] I have updated the documentation (wiki pages, ReadMe etc), if applies

### CODE REVIEW

File: a.txt [hunk 1]
```diff
@@ -0,0 +1,134 @@
+commit 5b04a8ac3458a9e25c65ba022adbbc41653cf8f7
```
Comment (buha) [inline a.txt:1]: remove a.txt file, i guess it was added by mistake

**Comment by buha** [comment]
i've looked at every commit and it looks good

File: a.txt (hunk 1 above)
Comment (ahaslam2) [reply a.txt:1]: ups, correct, this should not have been included..

**Comment by ahaslam2** [comment]
Changes V2:
- removed bad included file
- use macro for number of samples in basic example.

**Review by buha — APPROVED** [review]

**Pull request merged**